source/tool/chipsec/helper/linux/helper.py

```diff
@@ -158,12 +184,22 @@ def __mem_block(self, sz, newval = None):
         return 1
 
     def mem_read_block(self, addr, sz):
-        if(addr != None): self.dev_fh.seek(addr)
-        return self.__mem_block(sz)
+        if self.driver_loaded:
+            if(addr != None): self.dev_fh.seek(addr)
+            return self.__mem_block(sz)
+        elif self.fallback_available():
+            os.lseek(self.dev_mem, addr, os.SEEK_SET)
+            return os.read(self.dev_mem, sz)
 
     def mem_write_block(self, addr, sz, newval):
-        if(addr != None): self.dev_fh.seek(addr)
-        return self.__mem_block(sz, newval)
+        if self.driver_loaded:
+            if(addr != None): self.dev_fh.seek(addr)
+            return self.__mem_block(sz, newval)
+        elif self.fallback_available():
+            os.lseek(self.dev_mem, addr, os.SEEK_SET)
+            written = os.write(self.dev_mem, newval)
+            if written != sz:
+                logger().error("Cannot write %s to memory %016x" % (newval, addr))
```

Comment:
include sz?
  logger().error("Cannot write %s to memory %016x (wrote %d of %d)" % (newval, addr, written, sz))